src/ORM/FieldType/DBEmail.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace SilverStripe\ORM\FieldType;
+
+use SilverStripe\ORM\FieldType\DBVarchar;
+use Symfony\Component\Validator\Constraints;
+use SilverStripe\Core\Validation\ConstraintValidator;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\EmailField;
+use SilverStripe\Forms\NullableField;
+use SilverStripe\Forms\FormField;
+
+class DBEmail extends DBVarchar
+{
+    public function validate(): ValidationResult
+    {
+        // https://symfony.com/doc/current/reference/constraints/Email.html
+        $result = parent::validate();
+        $message = _t('SilverStripe\\Forms\\EmailField.VALIDATION', 'Please enter an email address');
+        $result = $result->combineAnd(
+            ConstraintValidator::validate(
+                $this->getValue(),
+                new Constraints\Email(message: $message),
+                $this->getName()
+            )
+        );
+        $this->extend('updateValidate', $result);
+        return $result;
+    }
+
+    public function scaffoldFormField(?string $title = null, array $params = []): ?FormField
```

Comment:
This was copy pasted from DBVarchar, just changing the TextField to an EmailField